nano/slow_test/node.cpp

```diff
@@ -992,62 +932,46 @@ namespace transport
 		// The test waits until all telemetry_ack messages have been received.
 		for (int i = 0; i < num_threads; ++i)
 		{
-			threads.emplace_back ([&node_data_single, &node_data_all, &shared_data_single, &shared_data_all]() {
-				auto func = [](auto & all_node_data_a, shared_data & shared_data_a, bool single_a) {
-					while (std::any_of (all_node_data_a.cbegin (), all_node_data_a.cend (), [](auto const & data) { return data.keep_requesting_metrics.load (); }))
+			threads.emplace_back ([&node_data, &shared_data]() {
+				while (std::any_of (node_data.cbegin (), node_data.cend (), [](auto const & data) { return data.keep_requesting_metrics.load (); }))
+				{
+					for (auto & data : node_data)
 					{
-						for (auto & data : all_node_data_a)
+						// Keep calling get_metrics_async until the cache has been saved and then become outdated (after a certain period of time) for each node
+						if (data.keep_requesting_metrics)
 						{
-							// Keep calling get_metrics_async until the cache has been saved and then become outdated (after a certain period of time) for each node
-							if (data.keep_requesting_metrics)
-							{
-								++shared_data_a.count;
-
-								if (single_a)
-								{
-									// Pick first peer to be consistent
-									auto peer = data.node->network.tcp_channels.channels[0].channel;
-									data.node->telemetry.get_metrics_single_peer_async (peer, [&shared_data_a, &data, &all_node_data_a](nano::telemetry_data_response const & telemetry_data_response_a) {
-										callback_process (shared_data_a, data, all_node_data_a, *telemetry_data_response_a.telemetry_data.timestamp);
-									});
-								}
-								else
-								{
-									data.node->telemetry.get_metrics_peers_async ([&shared_data_a, &data, &all_node_data_a](nano::telemetry_data_responses const & telemetry_data_responses_a) {
-										callback_process (shared_data_a, data, all_node_data_a, *telemetry_data_responses_a.telemetry_datas.begin ()->second.timestamp);
-									});
-								}
-							}
-							std::this_thread::sleep_for (1ms);
+							++shared_data.count;
+
+							// Pick first peer to be consistent
+							auto peer = data.node->network.tcp_channels.channels[0].channel;
+							data.node->telemetry->get_metrics_single_peer_async (peer, [&shared_data, &data, &node_data](nano::telemetry_data_response const & telemetry_data_response_a) {
+								ASSERT_FALSE (telemetry_data_response_a.error);
+								callback_process (shared_data, data, node_data, *telemetry_data_response_a.telemetry_data.timestamp);
+							});
 						}
+						std::this_thread::sleep_for (1ms);
 					}
+				}
 
-					shared_data_a.shared_future.wait ();
-					ASSERT_EQ (shared_data_a.count, 0);
-				};
-
-				func (node_data_single, shared_data_single, true);
-				func (node_data_all, shared_data_all, false);
+				shared_data.shared_future.wait ();
+				ASSERT_EQ (shared_data.count, 0);
```

Comment:
Not necessary now, but it would be nice if we could make the necessary adjustments to make `nano::util::counted_completion` work here; I think it's close.

Reply:
Done